firewood/src/merkle.rs

```diff
@@ -1182,6 +1186,20 @@ impl<S: ShaleStore<Node> + Send + Sync> Merkle<S> {
     }
 }
 
+fn set_parent(new_chd: DiskAddress, parents: &mut [(ObjRef<'_, Node>, u8)]) {
```

Comment:
This seems so strange. We pass in a mutable array of parents, then immediately only get the last parent and modify it? Also idx (the second u8) is never mutated. Maybe:
`fn set_parent(new_chd: DiskAddress, mut parent: ObjRef<'_, Node>, parent_node_idx: u8)`?

Followup?

Reply:
Yeah, I looked at changing this, but honestly (for now), I think it's best if it stays the way that it is. `set_parent` is called inside that really awful `write_node` macro which is why I didn't change it. 

Step 1: get rid of the macro altogether (I'm not sure that it provides a ton of value)
Step 2: fix `set_parent`. 

Just my opinion... But yeah, I completely agree this is super weird, but to change this now would have been too big a change IMO